Remove test and extra sections from meta.yaml

tcmetzger opened the pull request:
While updating the [CI section](https://docs.bokeh.org/en/latest/docs/dev_guide/testing.html#continuous-integration) of the dev docs, I noticed some outdated information in `meta.yaml` (e.g. `sphinx >=1.8`).
This PR removes the sections `test` and `extra` that are no longer needed.

cc @bryevdv

bryevdv commented on conda.recipe/meta.yaml:
```diff
@@ -41,46 +41,6 @@ requirements:
-    - selenium >=3.8
-    - scipy
-
     # examples
     - flask >=1.0
     - django
```
these lines were part of the `test` stanza I think

bryevdv commented on conda.recipe/meta.yaml:
```diff
@@ -41,46 +41,6 @@ requirements:
     - {{ dep.lower() }}
     {% endfor %}
 
-test:
```
we may want to leave the top-level `test` stanza but delete everything under it except the `commands` and `imports` sub-stanzas. IIRC conda-build will use those to perform  very rudimentary smoke-tests on the built package.

bryevdv commented:
OTOH we currently build with `--no-test` so maybe there is no point to adding more test imports unless we want to change that too 

https://github.com/bokeh/bokeh/blob/branch-2.4/release/build.py#L60

tcmetzger commented:
Some info re my most recent commits:

* I expanded the `about` section to include a `description` (the intro text we use at https://docs.bokeh.org/en/latest/#bokeh-documentation). I also added  `dev_url` (pointing to the GitHub URL we defined as `url` in setup.py) and `doc_url`. I changed `home` to point to www.bokeh.org instead of the GitHub repo. 
* I removed the entire `test` section, as suggested. As far as I understand, it makes sense to leave testing to the CI/maual pre-release testing, but I don't know enough about the conda ecosystem to make a final decision here. 
* In setup.py, I updated the `url` from `http` to `https`.

bryevdv commented:
LGTM
